src/corefx/System.Globalization.Native/icushim.cpp

```diff
@@ -170,6 +170,13 @@ bool FindLibWithMajorMinorSubVersion(int* majorVer, int* minorVer, int* subVer)
     return false;
 }
 
+static int32_t g_icuPresent = 0;
+
+extern "C" int32_t GlobalizationNative_ICUPresent()
+{
+    return g_icuPresent;
```

Comment:
So this means that we will *always* load ICU if there is one available, and then not use it. It is wasteful to load 20MB binary that we are going to not use, and makes it hard to verify the correctness by construction.

This should be changed so that the ICU is loaded only when the globalization is invariant globalization is not enabled.